cpp/src/common/cumlHandle.hpp

```diff
@@ -25,51 +25,59 @@
 #include <cusolverDn.h>
 #include <cusparse.h>
 
-#include "../cuML.hpp"
+#include "common/cuml_comms_int.hpp"
+
 #include "../cuML_api.h"
+#include "../cuML.hpp"
 
 namespace ML {
 
 /**
  * @todo: Add doxygen documentation
  */
 class cumlHandle_impl {
- public:
-  cumlHandle_impl();
-  ~cumlHandle_impl();
-  int getDevice() const;
-  void setStream(cudaStream_t stream);
-  cudaStream_t getStream() const;
-  void setDeviceAllocator(std::shared_ptr<deviceAllocator> allocator);
-  std::shared_ptr<deviceAllocator> getDeviceAllocator() const;
-  void setHostAllocator(std::shared_ptr<hostAllocator> allocator);
-  std::shared_ptr<hostAllocator> getHostAllocator() const;
-
-  cublasHandle_t getCublasHandle() const;
-  cusolverDnHandle_t getcusolverDnHandle() const;
-  cusparseHandle_t getcusparseHandle() const;
-
-  cudaStream_t getInternalStream(int sid) const;
-  int getNumInternalStreams() const;
-
-  void waitOnUserStream() const;
-  void waitOnInternalStreams() const;
-
- private:
-  //TODO: What is the right number?
-  static constexpr int _num_streams = 3;
-  const int _dev_id;
-  std::vector<cudaStream_t> _streams;
-  cublasHandle_t _cublas_handle;
-  cusolverDnHandle_t _cusolverDn_handle;
-  cusparseHandle_t _cusparse_handle;
-  std::shared_ptr<deviceAllocator> _deviceAllocator;
-  std::shared_ptr<hostAllocator> _hostAllocator;
-  cudaStream_t _userStream;
-  cudaEvent_t _event;
-
-  void createResources();
-  void destroyResources();
+public:
```

Comment:
Unnecessary white space changes due to not applying formatting rules.

Reply:
This actually happened in a merge and I did not want to the formatter on the merged files before committing them because it would have made many more deltas. I will clean this one up.